ttnn/cpp/ttnn/operations/data_movement/tilize_with_val_padding/device/tilize_with_val_padding_program_factory.cpp

```diff
@@ -235,11 +247,26 @@ operation::ProgramWithCallbacks tilize_with_val_padding_multi_core_interleaved(
     uint32_t stick_size_is_power_of_two = is_power_of_two_at_least_32(stick_size);
     uint32_t log2_stick_size = stick_size_is_power_of_two ? (std::uint32_t)std::log2(stick_size) : 0;
 
+    uint32_t shift_bits;
+    uint32_t packed_pad_value;
+    if (a.get_dtype() == DataType::BFLOAT16) {
+        bfloat16 bfloat_pad_value = bfloat16(std::get<float>(pad_value));
+        packed_pad_value = pack_two_bfloat16_into_uint32({bfloat_pad_value, bfloat_pad_value});
+        shift_bits = 6;
+    }
+    //UINT32
+    else {
+        TT_FATAL(a.get_dtype() == DataType::UINT32, "data type must be bfloat16 or uint32");
```

Comment:
nit: update commnet to mentionn just UINT32.
I think this fatal is enough and that you can remove comment above //UINT32
I would do the same thing on line 127